### PR TITLE
check market protocol before making boost deal 

### PR DESCRIPTION
**Pull request by LexLuthr**

### CODE REVIEW

File: cmd/boost/deal_cmd.go [hunk 1]
```diff
@@ -149,6 +150,26 @@ func dealCmdAction(cctx *cli.Context, isOnline bool) error {
...
+	sort.Strings(protos)
+
+	var protoFound bool
+	for _, v := range protos {
+		if v == DealProtocolv120 {
+			protoFound = true
+			break
+		}
+		protoFound = false
+		continue
```
Comment (nonsense) [inline cmd/boost/deal_cmd.go:166]:
```suggestion
```

File: cmd/boost/deal_cmd.go [hunk 2]
```diff
@@ -149,6 +150,26 @@ func dealCmdAction(cctx *cli.Context, isOnline bool) error {
 		return fmt.Errorf("failed to connect to peer %s: %w", addrInfo.ID, err)
 	}
 
+	protos, err := n.Host.Peerstore().GetProtocols(addrInfo.ID)
+	if err != nil {
+		return fmt.Errorf("getting protocols for peer %s: %w", addrInfo.ID, err)
+	}
+	sort.Strings(protos)
```
Comment (nonsense) [inline cmd/boost/deal_cmd.go:157]:
```suggestion
```

File: cmd/boost/deal_cmd.go [hunk 3]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"sort"
```
Comment (nonsense) [inline cmd/boost/deal_cmd.go:7]:
```suggestion
```

**Review by nonsense — APPROVED** [review]

File: cmd/boost/deal_cmd.go [hunk 4]
```diff
@@ -149,6 +149,24 @@ func dealCmdAction(cctx *cli.Context, isOnline bool) error {
...
+		return fmt.Errorf("getting protocols for peer %s: %w", addrInfo.ID, err)
+	}
+
+	var protoFound bool
+	for _, v := range protos {
+		if v == DealProtocolv120 {
+			protoFound = true
+			break
+		}
+		protoFound = false
```
Comment (nonsense) [inline cmd/boost/deal_cmd.go:163]:
```suggestion
```

**Comment by dirkmc** [comment]
I think we can just use `n.Host.Peerstore().FirstSupportedProtocol()`

**Review by dirkmc — APPROVED** [review]

**Comment by dirkmc** [comment]
Looks good, thanks 👍

**Pull request merged**